kernel/rtlil.h

```diff
@@ -796,11 +815,10 @@ struct RTLIL::Const
 		bv.resize(width, bv.empty() ? RTLIL::State::Sx : bv.back());
 	}
 
-	inline unsigned int hash() const {
-		unsigned int h = mkhash_init;
-
-		for (State b : *this)
-			h = mkhash(h, b);
+	inline Hasher hash_eat(Hasher h) const {
+		// TODO hash size
+		for (auto b : *this)
+			h.eat(b);
```

Comment:
We need a string represented as string vs as bit vector to have equal hashes, so we have to go bit by bit. To hash a contiguous chunk, for string represented Const, we'd need to first construct a new bit vector to run it on that. So it doesn't sound like it pays off

Reply:
Yeah, that's what I meant with "I'm assuming this would cause issues when we have two equal consts, one represented as states one as string". I have some ideas on how to compute a hash that processes more than a bit at a time while still agreeing between the two representations. That would be quite a bit more complex though, so probably only worth it if hashing consts is a significant overall cost. Do you happen to have any data on that specifically? In any case, this would be something we can do in a later PR.

Reply:
Ah sorry, I misread that. This is an interesting challenge. I haven't noticed hashing Consts as a hotspot so far